freeipa/src/main/java/com/sequenceiq/freeipa/kerberosmgmt/v1/KerberosMgmtV1Service.java

```diff
@@ -131,10 +131,16 @@ private boolean hostAdd(String hostname, FreeIpaClient ipaClient) {
             if (response == null || response.getResult() == null) {
                 return false;
             }
-            // TODO Handle Failure cases.
         } catch (FreeIpaClientException e) {
-            LOGGER.error(HOST_CREATION_FAILED + e.getMessage());
-            return false;
+            if (!Optional.ofNullable(e.getCause())
+                    .filter(JsonRpcClientException.class::isInstance)
+                    .map(JsonRpcClientException.class::cast)
+                    .map(JsonRpcClientException::getCode)
+                    .filter(c -> c == DUPLICATE_ENTRY_ERROR_CODE)
+                    .isPresent()) {
+                LOGGER.error(HOST_CREATION_FAILED + e.getMessage());
```

Comment:
if it's expected I don't think we should log this as error. Warn or even info should be enough

Reply:
In the normal execution paths where either the host was successfully created or the host already exists, this does not log anything. Line 135 has a `!` so the statement actually ignores duplicates and only logs error if there was a failure to create the host.

Reply:
ok, but then is it a good idea to not have the stacktrace? will be the message enough?
side note: could you add space or a delimiter to the log message, as it seems to me `HOST_CREATION_FAILED` doesn't end with space.

Reply:
is it a good idea to not have the stacktrace? Good point. I added it.

could you add space? Done